blog/2024/preview-keycloak-test-framework.adoc

```diff
@@ -0,0 +1,42 @@
+:title: "Writing new tests has never been easier than before." Keycloak Test Framework
+:date: 2024-08-28
+:publish: true
+:author: Lukas Hanusovsky
+
+== How It All Started
+
+The idea to replace the current https://github.com/keycloak/keycloak/tree/main/testsuite[test suite] has been on the table for multiple years. Initially, it was meant to be only a refactoring of the current approach on how to write tests, but after a few internal discussions and refactor updates it turned out a new test suite, based on a https://github.com/keycloak/keycloak/tree/main/test-framework[new framework] would be a better solution.
+
+It would be good to mention a few drawbacks, that stand out when working with the current https://github.com/keycloak/keycloak/tree/main/testsuite[test suite]. First of all, is the complexity of various configurations and additions made on top of the https://arquillian.org/[_Arquillian framework_]. These changes make the test suite powerful, but the cons is that without proper documentation for beginners is almost unreadable. The second thing has the same importance, the _Arquillian framework_ is not fully supported anymore. Other things to mention are a complicated execution system, where you want to specify what exactly should be tested, then abstract classes with shared configurations, missing the option to add a custom extension, and last but not least, the massive amount of tests without knowledge if they are needed anymore.
```

Comment:
Not convinced about the statement `the massive amount of tests without knowledge if they are needed anymore`. There may be some tests not needed, but probably not a massive amount. Perhaps just drop this?